morf-core/src/main/java/org/alfasoftware/morf/sql/SqlTokenizer.java

```diff
@@ -0,0 +1,707 @@
+        char c = query.charAt(i);
+
+        // end of the multi-line comment
+        if (c == '\n')
+          return i + 1;
+        if (c == '\r' && i+1 < length && query.charAt(i+1) == '\n')
+          return i + 2;
+        if (c == '\r')
+          return i + 1;
+
+        if (isAmbiguousNewlineCharacter(c)) {
+          commentCutShort.set(true);
+          return i; // stop the comment right here
+        }
+
+        i++;
+      }
+
+      return i;
+    }
+
+    return -1;
+  }
+
+
+  /**
+   * Tries to match a multi-line comment.
+   *
+   * Note: This currently matches hints, since they are effectively multi-line comments.
+   *
+   * See https://www.postgresql.org/docs/8.0/sql-syntax.html#SQL-SYNTAX-COMMENTS
+   * See https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Comments.html
+   *
+   * @param databaseType ORACLE/PGSQL
+   * @param query Query to be tokenized.
+   * @param length Total length of the query.
+   * @param i Position where this token matching should start.
+   * @param commentUnfinished This is set to true, when the comment is left unfinished before the query ends.
+   * @return End position of the token, if matched at this position, or negative number otherwise.
+   */
+  private static int matchMultiLineComment(String databaseType, String query, int length, int i, AtomicBoolean commentUnfinished) {
+    boolean allowCommentNesting = false;
+
+    if (PGSQL.equals(databaseType)) {
+      allowCommentNesting = true;
+    }
+
+    if (query.charAt(i) == '/' && i+1 < length && query.charAt(i+1) == '*') {
+      i += 2;
+      while (i < length) {
+        // end of the multi-line comment
+        if (query.charAt(i) == '*' && i+1 < length && query.charAt(i+1) == '/')
+          return i + 2;
+
+        if (allowCommentNesting) {
+          if (query.charAt(i) == '/' && i+1 < length && query.charAt(i+1) == '*') {
+            int mlc = matchMultiLineComment(databaseType, query, length, i, commentUnfinished);
+            if (mlc > 0)
+              i = mlc;
+          }
+        }
+
+        i++;
+      }
+
+      commentUnfinished.set(true);
+      return -1;
+    }
+
+    return -1;
+  }
+
+
+  /**
+   * See https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
+   * See https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Database-Object-Names-and-Qualifiers.html
+   */
+  private static boolean isIdentifierStartCharacter(char c) {
+    return 'A' <= c && c <= 'Z'
+        || 'a' <= c && c <= 'z'
+        || c == '_';
+  }
+
+
+  /**
+   * See https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
+   * See https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Database-Object-Names-and-Qualifiers.html
+   */
+  private static boolean isIdentifierMidCharacter(char c) {
+    return isIdentifierStartCharacter(c)
+        || '0' <= c && c <= '9'
+      //|| c == '#' // Oracle allows # in identifiers
```

Comment:
commented code

Reply:
That actually was intentional. I thought we should just disallow use of # in identifiers if we could.
But then it would have looked like an omission in the code, hence it was left included, but commented out.

Reply:
I will add a note to the comment making it more obvious